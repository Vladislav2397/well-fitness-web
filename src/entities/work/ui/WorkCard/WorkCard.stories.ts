import { Story } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

import WorkCard, { type WorkCardProps } from './WorkCard.vue'
import { defineControls } from '@/shared/lib/stories'

export default {
    title: 'undefined-path/WorkCard',
    component: WorkCard,
    argTypes: defineControls({
        //
    }),
}

const Template: Story<WorkCardProps> = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { WorkCard },
    methods: {
        handle: action('clicked'),
    },
    template: `<WorkCard v-bind="$props" @click="handle" />`,
})

export const Default = Template.bind({})
Default.args = {
    //
}
