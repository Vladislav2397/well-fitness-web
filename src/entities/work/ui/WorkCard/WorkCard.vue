<template lang="pug">

.b-work-card

</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'

export type WorkCardProps = {
    //
}

@Component
export default class WorkCard extends Vue {
    //
}
</script>

<!-- <style lang="scss" src="./WorkCard.critical.scss" /> -->
<!-- <style lang="scss" src="./WorkCard.main.scss" /> -->
