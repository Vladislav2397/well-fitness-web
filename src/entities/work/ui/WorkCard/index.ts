import WorkCard from './WorkCard.vue'

export { WorkCard }
