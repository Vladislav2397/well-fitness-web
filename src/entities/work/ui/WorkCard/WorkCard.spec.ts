import { shallowMount } from '@vue/test-utils'
import WorkCard /*, { type WorkCardProps }*/ from './WorkCard.vue'

describe('test WorkCard', () => {
    test('snapshot', () => {
        const wrapper = shallowMount(WorkCard, {
            propsData: {
                //
            } /* as WorkCardProps*/,
        })
        expect(wrapper).toMatchSnapshot()
    })
})
