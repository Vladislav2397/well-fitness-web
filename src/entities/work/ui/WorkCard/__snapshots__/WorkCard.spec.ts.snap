// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test WorkCard snapshot 1`] = `<div class="b-work-card"></div>`;
